### PR TITLE
Person Attribute Forms Advanced Conditional Fields Support

### PR DESCRIPTION
**Pull request by nateh777**
##### What does the change add or fix?

Add support for conditional fields in Person Attribute Forms Advanced block. Due to the way it saves the fields to the database it currently only supports Attributes for conditions. This is not that different than Registration template conditional fields as the only supported person field was "Gender" there, but we cannot even support that in this block.

---------

##### What does the change add or fix in a succinct statement that will be read by clients?

- Added support for Conditional Fields/Form Field Filters on Person Attributes.

---------

##### Who reported, requested, or paid for the change?

OHC

---------

##### Does this update or add options to the block UI?

image

image

---------

##### What files does it affect?

- Crm/PersonAttributeForms.ascx
- Crm/PersonAttributeForms.ascx.cs

---------

##### Is it a breaking change for other versions/clients?

No, there was one change to Guid's of fields, but steps were taken to make it backwards compatible when conditions are added.

### CODE REVIEW

File: Crm/PersonAttributeForms.ascx.cs [hunk 1]
```diff
@@ -1037,6 +1052,22 @@ private void tfeForm_DeleteFieldClick( object sender, AttributeFormFieldEventArg
                 var field = form.Fields.FirstOrDefault( f => f.Guid == e.FormFieldGuid );
                 if ( field != null )
                 {
+                    /*
+                      On Field Delete, we need to also remove all the existing reference of current field in filter rule list
+                    */
+                    var newFormFieldsWithRules = form.Fields
```
Comment (nateh777) [inline Crm/PersonAttributeForms.ascx.cs:1058]: Test functionality within 13 to ensure nothing changed

File: Crm/PersonAttributeForms.ascx.cs [hunk 2]
```diff
@@ -1436,13 +1582,16 @@ private void BuildViewControls( bool setValues )
                                     }
                                 }
                             }
+
                         }
 
                         if ( !string.IsNullOrWhiteSpace( field.PostText ) )
                         {
                             phContent.Controls.Add( new LiteralControl( field.PostText.ResolveMergeFields( mergeFields ) ) );
                         }
```
Comment (nateh777) [inline Crm/PersonAttributeForms.ascx.cs:1591]:
```suggestion
```

File: Crm/PersonAttributeForms.ascx.cs [hunk 3]
```diff
@@ -1397,6 +1508,7 @@ private void BuildViewControls( bool setValues )
                             phContent.Controls.Add( new LiteralControl( field.PreText.ResolveMergeFields( mergeFields ) ) );
```
Comment (nateh777) [inline Crm/PersonAttributeForms.ascx.cs:1508]: Remove pretext code as it is done with the field visibility wrapper now.

File: Crm/PersonAttributeForms.ascx.cs (hunk 1 above)
Comment (nateh777) [reply Crm/PersonAttributeForms.ascx.cs:1058]: Tested in v13 and did a quick cursory review of some of the code, doesn't look like it changed at all.

**Review by mattbaylor — APPROVED** [review]

**Pull request merged**